### PR TITLE
:wrench: Updated modal in prescription

### PR DESCRIPTION
**Pull request by issa-diallo**
- Create a patient with required fields only

### CODE REVIEW

File: src/components/modal/ModalAddPatient.tsx [hunk 1]
```diff
@@ -57,45 +57,6 @@ const ModalAddPatient: React.FC<ModalAddPatientProps> = ({
...
-
-          <Form.Group as={Col} className='mt-2'>
-            <Form.Label><FontAwesomeIcon icon={['fas', 'city']} />  {t('form.zipPostal')}</Form.Label>
-            <Form.Control
-              name='zip_code'
-              type='text'
-              value={patientState.zip_code}
-              onChange={handleChangeNewPatient}
-            />
-          </Form.Group>
```
Comment (AndreMiras) [inline src/components/modal/ModalAddPatient.tsx:98]: Why not implementing what we suggested using collapse?
https://react-bootstrap-v4.netlify.app/components/accordion/

Comment (issa-diallo) [reply src/components/modal/ModalAddPatient.tsx:98]: I will create a ticket for it because it is not urgently

Comment (issa-diallo) [reply src/components/modal/ModalAddPatient.tsx:98]: https://github.com/mynotif/mynotif-frontend/issues/235

File: src/components/forms/PatientForm.tsx [hunk 2]
```diff
@@ -23,7 +23,7 @@ const PatientForm: FunctionComponent<PatientFormProps> = ({ patient, isEditForm
   const { t } = useTranslationHook()
   const navigate = useNavigate()
 
-  const birthdayDateValue: Date | null = patient.birthday !== '' ? new Date(patient.birthday) : null
+  const birthdayDateValue: Date | null = patient.birthday !== null && patient.birthday !== undefined && patient.birthday !== '' ? new Date(patient.birthday) : null
```
Comment (AndreMiras) [inline src/components/forms/PatientForm.tsx:26]: Would this work?
```
const birthdayDateValue: Date | null = patient.birthday ? new Date(patient.birthday) : null;
```

File: src/components/forms/PrescriptionForm.tsx [hunk 3]
```diff
@@ -144,11 +144,23 @@ const PrescriptionForm: FunctionComponent<PrescriptionFormProps> = ({
     }
   }
 
+  const updatePatientDate = (date: Date, field: string): void => {
+    const formattedDate = format(date, 'yyyy-MM-dd')
```
Comment (AndreMiras) [inline src/components/forms/PrescriptionForm.tsx:148]: DRY date format also used in src/components/forms/PatientForm.tsx

File: src/components/modal/ModalAddPatient.tsx [hunk 4]
```diff
@@ -17,11 +20,19 @@ const ModalAddPatient: React.FC<ModalAddPatientProps> = ({
   patientState,
   handleChangeNewPatient,
   handleNewPatientSubmit,
+  updatePatientDate,
   onHide,
   show,
   error
 }) => {
   const { t } = useTranslationHook()
+  const birthdayDateValue: Date | null = patientState.birthday !== null && patientState.birthday !== undefined && patientState.birthday !== '' ? new Date(patientState.birthday) : null
```
Comment (AndreMiras) [inline src/components/modal/ModalAddPatient.tsx:29]:
```
const birthdayDateValue: Date | null = patient.birthday ? new Date(patient.birthday) : null;
```

File: src/components/modal/ModalAddPatient.tsx [hunk 5]
```diff
@@ -56,46 +66,82 @@ const ModalAddPatient: React.FC<ModalAddPatientProps> = ({
...
+                    value={patientState.street}
+                    onChange={handleChangeNewPatient}
+                  />
+                </Form.Group>
+                <Form.Group as={Col} className='mt-2'>
+                  <Form.Label className='me-2'><FontAwesomeIcon icon={['fas', 'calendar-days']} /> {t('form.birthday')}</Form.Label>
+                  <ReactDatePicker
+                    selected={birthdayDate}
+                    onChange={onBirthdayChange}
+                    dateFormat='dd/MM/yyyy'
```
Comment (AndreMiras) [inline src/components/modal/ModalAddPatient.tsx:88]: reuse from a const to be defined somewhere

File: src/components/forms/PatientForm.tsx (hunk 2 above)
Comment (issa-diallo) [reply src/components/forms/PatientForm.tsx:26]: no I have to check `Unexpected nullable string value in conditional.`

Comment (AndreMiras) [reply src/components/forms/PatientForm.tsx:26]: OK linter wants it explicit which makes sense but it's painful in this case

File: src/components/forms/PrescriptionForm.tsx (hunk 3 above)
Comment (issa-diallo) [reply src/components/forms/PrescriptionForm.tsx:149]: done

File: src/components/modal/ModalAddPatient.tsx (hunk 4 above)
Comment (issa-diallo) [reply src/components/modal/ModalAddPatient.tsx:30]: done

File: src/components/modal/ModalAddPatient.tsx (hunk 5 above)
Comment (issa-diallo) [reply src/components/modal/ModalAddPatient.tsx]: done

File: src/components/forms/PrescriptionForm.tsx (hunk 3 above)
Comment (AndreMiras) [reply src/components/forms/PrescriptionForm.tsx:149]: Use `STANDARD_DATE_FORMAT` here

File: src/components/modal/ModalAddPatient.tsx (hunk 4 above)
Comment (AndreMiras) [reply src/components/modal/ModalAddPatient.tsx:30]: Not really, but agreed to not address to keep the linter happ

File: src/services/constants.ts [hunk 6]
```diff
@@ -1,8 +1,11 @@
 import assert from 'assert'
 
-const BACKEND_URL: string = process.env.REACT_APP_BACKEND_URL ?? 'http://localhost:8000'
+const BACKEND_URL: string = 'http://localhost:8000'
```
Comment (AndreMiras) [inline src/services/constants.ts:3]: Why did this change?

File: src/services/constants.ts [hunk 7]
```diff
@@ -1,8 +1,11 @@
...
 assert(typeof BACKEND_URL === 'string')
 
+const STANDARD_DATE_FORMAT: string = ' dd/MM/yyyy'
+
 export {
-  BACKEND_URL
+  BACKEND_URL,
+  STANDARD_DATE_FORMAT
```
Comment (AndreMiras) [inline src/services/constants.ts:10]: Would our code formatter accept trailing comma? That way we avoid 2 lines diff for one line change

**Review by AndreMiras — APPROVED** [review]

**Review by AndreMiras — APPROVED** [review]

**Pull request merged**